src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -132,20 +131,20 @@ def get_counts_for_project(self, project_id: int) -> Iterable[base.BucketedCount
         This may throw an exception if there is some sort of issue fetching counts from the redis
         store.
         """
-        key_prefix = f"{self._counter_key_prefix()}:{project_id}:"
+        bucket_size = self._counter_bucket_size
+        now = int(time.time())
+        now_timestamp = now - now % bucket_size
 
-        keys = sorted(
-            self.cluster.scan_iter(
-                match=key_prefix + "*",
-            )
-        )
-        counts = self.cluster.mget(keys)
-        for key, count_raw in zip(keys, counts):
-            _, timestamp_raw = key.split(key_prefix)
+        first_timestamp = int(now - self._counter_ttl / datetime.timedelta(seconds=1))
```

Comment:
i believe you can use `self._counter_ttl.total_seconds()` in place of `self._counter_ttl / datetime.timedelta(seconds=1)` here. `total_seconds()` does return a float though!